From: 	Emmanuel Grumbach <emmanuel.grumbach@intel.com>
Date: 	Tue, 28 Oct 2014 08:11:26 +0200
Subject: iwlwifi: mvm: update layout of firmware error dump
Content-Transfer-Encoding: 8bit
Origin: https://git.kernel.org/linus/5bfe6f53283de44855ee45a102210abbfac995f9
Bug-Debian: https://bugs.debian.org/767088

The memory was not zeroed - fix that. Also update the
iwl_fw_error_dump_info structure.

Signed-off-by: Emmanuel Grumbach <emmanuel.grumbach@intel.com>
---
 drivers/net/wireless/iwlwifi/mvm/mac80211.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/iwlwifi/mvm/mac80211.c b/drivers/net/wireless/iwlwifi/mvm/mac80211.c
index ed10e55..83ab42a 100644
--- a/drivers/net/wireless/iwlwifi/mvm/mac80211.c
+++ b/drivers/net/wireless/iwlwifi/mvm/mac80211.c
@@ -678,7 +678,7 @@ static void iwl_mvm_fw_error_dump(struct iwl_mvm *mvm)
 	if (trans_len)
 		file_len += trans_len;
 
-	dump_file = vmalloc(file_len);
+	dump_file = vzalloc(file_len);
 	if (!dump_file)
 		return;
 
